Sources/PointFree/Account/Invoices.swift

```diff
@@ -0,0 +1,302 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+// MARK: Middleware
+
+let invoicesResponse =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
+    <<< requireStripeSubscription
+    <<< fetchInvoices
+    <| writeStatus(.ok)
+    >-> map(lower)
+    >>> respond(
+      view: invoicesView,
+      layoutData: { subscription, invoicesEnvelope, currentUser, subscriberState in
+        SimplePageLayoutData(
+          currentSubscriberState: subscriberState,
+          currentUser: currentUser,
+          data: (subscription, invoicesEnvelope, currentUser),
+          title: "Payment history"
+        )
+    }
+)
+
+let invoiceResponse =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
+    <<< requireStripeSubscription
+    <<< filterMap(
+      over3(fetchInvoice) >>> sequence3 >>> map(require3),
```

Comment:
do we need an additional filter to make sure that the invoice belongs to the user? it seems that any invoice id could be provided and we'd just fetch it?

Reply:
Ah yeah. I originally thought that the customer was needed to form the request and forgot to come back to this! Good catch!